### PR TITLE
samba: Migrate add-on layout to S6 Overlay

### PR DESCRIPTION
**Pull request by lmagyar**
No functional change.

Just migrates files to new S6 layout.

I saw a planned update to Alpine 3.17 and thought it's a good time to change this also.

### CODE REVIEW

**Comment by frenck** [comment]
I have this + a bunch of more changes ready to go... while I was awaiting #2997.

This PR is running through my planned maintenance.

../Frenck

**Comment by lmagyar** [comment]
OK, closing, no problem. :)
Made another small PR, no more PR planned from me.

**Comment by frenck** [comment]
I actually have this and more changes already pending, you are now running through my samba add-on maintenance which I started in #

**Comment by frenck** [comment]
I don't think you need to close this PR? Like... I will be re-opening the same one 🤷 

(efforts have already been duplicated, that is all).

**Comment by lmagyar** [comment]
OK, I simply didn't want to disturb you. I reopened it and made it draft like the other #3002.

I will wait for the next samba release before I click anything. :)

**Comment by lmagyar** [comment]
Can be reopened, github actions bot didn't send notification about the stale label on this draft PR, only when closed it.

File: samba/rootfs/etc/s6-overlay/s6-rc.d/smbd/finish [hunk 1]
```diff
@@ -0,0 +1,26 @@
+#!/command/with-contenv bashio
+# shellcheck shell=bash
+# ==============================================================================
+# Take down the S6 supervision tree when smbd fails
+# ==============================================================================
+declare exit_code
```
Comment (frenck) [inline samba/rootfs/etc/s6-overlay/s6-rc.d/smbd/finish:6]: Unused variable

```suggestion
```

File: samba/rootfs/etc/s6-overlay/s6-rc.d/nmbd/finish [hunk 2]
```diff
@@ -0,0 +1,26 @@
+#!/command/with-contenv bashio
+# shellcheck shell=bash
+# ==============================================================================
+# Take down the S6 supervision tree when nmbd fails
+# ==============================================================================
+declare exit_code
```
Comment (frenck) [inline samba/rootfs/etc/s6-overlay/s6-rc.d/nmbd/finish:6]:
```suggestion
```

**Review by frenck — CHANGES_REQUESTED** [review]

File: samba/config.yaml [hunk 3]
```diff
@@ -1,5 +1,5 @@
 ---
-version: 11.0.0
+version: 11.1.0
```
Comment (frenck) [inline samba/config.yaml:2]:
```suggestion
version: 11.0.0
```

Comment (frenck) [reply samba/config.yaml:2]: Adding this one, to avoid the upgrade/release. Got another PR inbound for the 11.1.0

**Review by frenck — APPROVED** [review]
Thanks, @lmagyar 👍 

../Frenck

**Pull request merged**